docs/guides/push_to_deploy_on_github.md

```diff
@@ -0,0 +1,147 @@
+we will call `shorebird_push_to_deploy`.
+
+## 🚀 Getting Started
+
+First thing, lets create a new Flutter project simply by running:
+
+```bash
+flutter create shorebird_push_to_deploy
+```
+
+Be sure to initialize your project as a git repository:
+
+```bash
+cd shorebird_push_to_deploy
+git init
+git remote add origin <THE_URL_OF_YOUR_REPOSITORY>
+```
+
+Next we need to initialize shorebird in it! If you have doubts on how to do so, check the
+[Code Push Getting Started](/code_push/initialize) guide.
+
+## 🛠️ Setting up GitHub Actions
+
+We provide a collection of official Github Actions to help you with the integration. For this guide
+we will be using `shorebird-setup`, `shorebird-release` and `shorebird-patch`, you can check more
+details about them at our [Github Integration Documentation](ci/github).
+
+Create a new file at `.github/workflows/push_to_deploy_android.yml` with the following content:
+
+```yaml
+name: Shorebird Android
+
+on:
+  push:
+    branches:
+      - releases/*
+
+env:
+  SHOREBIRD_TOKEN: ${{ secrets.SHOREBIRD_TOKEN }}
+
+jobs:
+  push_to_deploy_android:
+    defaults:
+      run:
+        shell: bash
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: 📚 Git Checkout
+        uses: actions/checkout@v3
+
+      - name: 🐦 Setup Shorebird
+        uses: shorebirdtech/setup-shorebird@v1
+
+      - name: 🚀 Shorebird Release
+        if: ${{ github.event.created }}
+        uses: shorebirdtech/shorebird-release@v0
+        with:
+          platform: android
+
+      - name: 🚀 Shorebird Patch
+        if: ${{ !github.event.created }}
+        uses: shorebirdtech/shorebird-patch@v0
+        with:
+          platform: android
+```
+
+The content of this file is pretty straightforward. It defines a workflow that will be triggered
+whenever a push is made to a branch that starts with `releases/`.
+
+It will then run the clone of the repository and setup the Shorebird CLI.
+
+Next the workflow is divided into two conditional steps. If the push is the one that created the
+branch, it will trigger a release. Otherwise it will trigger a patch.
+
+Note that the `SHOREBIRD_TOKEN` is being used as an environment variable. This is a secret that
+should be stored in your repository settings. You might have done that already if you checked the
+[GitHub Integration Documentation](/ci/github) mentioned above, but if not be sure to check it
+and configure the authentication in the action for your Shorebird account.
+
+If all went right, you should be able to push the changes to your repository and see the workflow
+running at the Actions tab of your repository.
+
+Give it a try! Create a new branch called `releases/1.0.0` and push it to your repository. Then
+once the workflow finishes, check the Shorebird dashboard and you should see a new release there.
+
+Next you can try pushing a new commit to the same branch and see a patch being created 🎉.
```

Comment:
Maybe we should also include an example of a small code change (like changing the color similar to the quick start guide) and also include the git commands used.

Reply:
Added, lmk if that is what you had in mind